explorer/src/pages/Mixnodes/index.tsx

```diff
@@ -46,24 +49,58 @@ const columns = [
 
 export const PageMixnodes: React.FC = () => {
   const { mixnodes } = useContext(MainContext);
-  const [rows, setRows] = React.useState();
+  const [filteredMixnodes, setFilteredMixnodes] = React.useState<MixNodeResponse | undefined>()
+  const [pageSize, setPageSize] = React.useState<string>("50");
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const st = event.target.value;
+    if (st === '' && mixnodes?.data) {
+      const formatted = mixnodeToGridRow(mixnodes?.data)
+      setFilteredMixnodes(formatted)
+    } else {
+      const filtered = mixnodes?.data && mixnodes?.data.filter((m) => {
```

Comment:
instead of && use the optional chaining operator again

`mixnodes?.data?.filter`

Reply:
Optional chaining has been a bit playful for me. Will show you an example when I next encounter it.

Reply:
As discussed in priv chat, Optional Chaining issue killed; learned that u can add the O/Ch just before .filter() and .map() too. So everyday's a school day.